tests/unit/test_grid.py

```diff
@@ -21,25 +19,175 @@
     compute_horizontal_source_grids,
     compute_num_elements,
     compute_projected_horizontal_source_grids,
-    compute_source_swath,
     compute_target_area,
-    crs_from_source_data,
+    create_area_definition_for_projected_source_grid,
+    create_target_area_from_source,
     dims_are_lon_lat,
     dims_are_projected_x_y,
+    get_area_definition_from_message,
     grid_height,
     grid_width,
 )
 
 
-@patch('harmony_regridding_service.grid.AreaDefinition', wraps=AreaDefinition)
-def test_compute_target_area(mock_area):
+@patch(
+    'harmony_regridding_service.grid.create_target_area_from_source',
+    wraps=create_target_area_from_source,
+)
+@patch(
+    'harmony_regridding_service.grid.get_area_definition_from_message',
+    wraps=get_area_definition_from_message,
+)
+def test_compute_target_area_with_parameters(
+    mock_get_area_definition_from_message,
+    mock_create_target_area_from_source,
+    smap_projected_netcdf_file,
+    var_info_fxn,
+):
     """Ensure Area Definition correctly generated."""
```

Comment:
It might be worth a bit extra of an explanation in this doc-string now that there are more options (e.g., that the grid information is coming from the Harmony message parameters, not the source granule)

Reply:
c137e09